server/server.go

```diff
@@ -41,13 +55,15 @@ func registerUser(w http.ResponseWriter, res *http.Request) {
 	//temporarily print password, so that its not an unused variable
 	fmt.Printf("Encrypted Password: %v\n", passDigest)
 
-	w.Write([]byte("User Created"))
 	data.Users = append(data.Users, &newUser)
 
-	for _, v := range data.Users {
-		fmt.Printf("User: %v\n", *v)
+	jwtToken, signingErr := createJWT(newUser.Username)
+	if signingErr != nil {
+		respondWithError(w, "Internal Server Error", http.StatusInternalServerError)
 	}
 
+	sendJWT(w, jwtToken)
+
 }
 
 // Handle Login ==============================================================
```

Comment:
I just noticed this function parses a `RegisterBody` from the request body. However, a `/login` request should only contains the username and password as a `UserCredentials` object. You haven't defined this struct in the backend, but you can check the documentation for the `/login` endpoint to see what I'm referring to.

Reply:
docs: https://scratched-ethernet-1a9.notion.site/Documentation-ddd34b987fe245e08dc405983878e8ee